### PR TITLE
ISPN-9008 RpcManager.invokeCommandOnAll ignores cache member missing from cluster view

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-9008

I added an `ImmutableHopscotchHashSet` mainly because I didn't want to have both a `unmodifiableSet()` wrapper and `HashSet` wrapping `HashMap`. Considering addresses are random numbers I don't think hopscotch hashing per se can help with performance, but maybe we could do something to optimize `containsAll` @rvansa?

### CODE REVIEW

**Comment by danberindei** [comment]
Rebased. I ended up using `ImmutableHopscotchHashSet` only in JGroupsTransport, and there's no usage of `containsAll` any more.

**Comment by tristantarrant** [comment]
I believe https://ci.infinispan.org/job/Infinispan/job/PR-5909/4/testReport/junit/org.infinispan.jmx/RpcManagerMBeanTest/testSuccessRatio/ is related

**Comment by danberindei** [comment]
Indeed, I updated the test to use `MagicKey` for more determinism.

**Comment by danberindei** [comment]
There was one more test failure caused by a conflict with the ISPN-9062 fix, it's solved now.

**Comment by rvansa** [comment]
@danberindei I don't see how you'd like to optimize `containsAll`... Somehow sorting the hashes and scanning ranges in the map?

File: core/src/test/java/org/infinispan/replication/ForceSyncAsyncFlagsTest.java [hunk 1]
```diff
@@ -55,9 +52,7 @@ public void testForceAsyncFlagUsage() throws Exception {
 
       // check that the replication call was sync
       cache1.put("k", "v");
-      verify(mockTransport)
-            .invokeCommandOnAll(any(ReplicableCommand.class), any(ResponseCollector.class), any(DeliverOrder.class),
-                                anyLong(), any(TimeUnit.class));
+      verify(mockTransport).invokeCommandOnAll(any(), any(), any(), any(), anyLong(), any());
```
Comment (rvansa) [inline core/src/test/java/org/infinispan/replication/ForceSyncAsyncFlagsTest.java]: why have you removed the classes?

Comment (danberindei) [reply core/src/test/java/org/infinispan/replication/ForceSyncAsyncFlagsTest.java]: I just find it more readable when the method call fits into one line, and the test only cares about the name of the method being called (`invokeCommandOnAll` vs. `sendToAll`).

**Comment by rvansa** [comment]
LGTM, integrated.

**Pull request merged**